packages/shared-metrics/src/name.js

```diff
@@ -20,24 +20,26 @@ exports.payloadPrefix = 'name';
 // @ts-ignore
 exports.currentPayload = undefined;
 
-const MAX_ATTEMPTS = 60;
-const DELAY = 1000;
+exports.MAX_ATTEMPTS = 10;
+exports.DELAY = 1000;
 let attempts = 0;
 
 exports.activate = function activate() {
   attempts++;
+
   applicationUnderMonitoring.getMainPackageJsonStartingAtMainModule((err, packageJson) => {
     if (err) {
       return logger.warn('Failed to determine main package json. Reason: ', err.message, err.stack);
-    } else if (!packageJson && attempts < MAX_ATTEMPTS) {
+    } else if (!packageJson && attempts < exports.MAX_ATTEMPTS) {
       logger.debug('Main package.json could not be found. Will try again later.');
-      setTimeout(exports.activate, DELAY).unref();
+      setTimeout(exports.activate, exports.DELAY).unref();
       return;
     } else if (!packageJson) {
-      if (process.mainModule) {
+      if (require.main) {
```

Comment:
This section was not covered in our tests. Added a test. Made the attempts & delay dirtily configureable via `exports` and `reset` fn. No need to add ENV variables for this IMO